src/box/sql/build.c

```diff
@@ -1125,34 +1155,49 @@ vdbe_emit_ck_constraint_create(struct Parse *parser,
 	struct Vdbe *v = sqlGetVdbe(parser);
 	assert(v != NULL);
 	/*
-	 * Occupy registers for 5 fields: each member in
-	 * _ck_constraint space plus one for final msgpack tuple.
+	 * Occupy registers for 20 fields: each member in _func space plus one
+	 * for final msgpack tuple.
 	 */
-	int ck_constraint_reg = sqlGetTempRange(parser, 7);
-	sqlVdbeAddOp2(v, OP_SCopy, reg_space_id, ck_constraint_reg);
-	sqlVdbeAddOp4(v, OP_String8, 0, ck_constraint_reg + 1, 0,
+	int regs = sqlGetTempRange(parser, 20);
+	sqlVdbeAddOp2(v, OP_NextSystemSpaceId, BOX_FUNC_ID, regs);
+	sqlVdbeAddOp2(v, OP_Integer, effective_user()->uid, regs + 1);
+	char *func_name = sqlMPrintf(db, "check_%s_%s", space_name,
+				     ck_def->name);
+	sqlVdbeAddOp4(v, OP_String8, 0, regs + 2, 0, func_name, P4_DYNAMIC);
+	sqlVdbeAddOp2(v, OP_Integer, 0, regs + 3);
+	sqlVdbeAddOp4(v, OP_String8, 0, regs + 4, 0, "SQL_EXPR", P4_STATIC);
+	char *body = sqlDbStrDup(db, ck_def->expr_str);
+	sqlVdbeAddOp4(v, OP_String8, 0, regs + 5, 0, body, P4_DYNAMIC);
+	sqlVdbeAddOp4(v, OP_String8, 0, regs + 6, 0, "function", P4_STATIC);
+	char *param_list = sqlDbMallocRawNN(db, 32);
+	char *buf = mp_encode_array(param_list, 0);
+	sqlVdbeAddOp4(v, OP_Blob, buf - param_list, regs + 7,
+		      SQL_SUBTYPE_MSGPACK, param_list, P4_DYNAMIC);
+	sqlVdbeAddOp4(v, OP_String8, 0, regs + 8, 0, "any", P4_STATIC);
+	sqlVdbeAddOp4(v, OP_String8, 0, regs + 9, 0, "none", P4_STATIC);
```

Comment:
This could look more understandable if you would use the names from schema_def.h:
```C
sqlVdbeAddOp4(v, OP_String8, 0, regs + BOX_FUNC_FIELD_RETURNS, 0, "any", P4_STATIC);
sqlVdbeAddOp4(v, OP_String8, 0, regs + BOX_FUNC_FIELD_AGGREGATE, 0, "none", P4_STATIC);
// ...
```
Up to you.

Reply:
Fixed.